app/src/main/java/com/spoony/spoony/core/designsystem/component/image/SpoonyImage.kt

```diff
@@ -0,0 +1,51 @@
+package com.spoony.spoony.core.designsystem.component.image
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.Image
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+
+@Composable
+fun SpoonyImage(
+    imageUrl: String,
+    modifier: Modifier = Modifier,
+    shape: Shape = RectangleShape,
+    contentScale: ContentScale = ContentScale.Crop,
+    contentDescription: String? = null
+) {
+    AsyncImage(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(imageUrl)
+            .crossfade(true)
+            .build(),
+        contentDescription = contentDescription,
+        contentScale = contentScale,
+        modifier = modifier
+            .clip(shape)
+    )
+}
+
+@Composable
+fun SpoonyImage(
+    @DrawableRes imageId: Int,
+    modifier: Modifier = Modifier,
+    shape: Shape = RectangleShape,
+    contentScale: ContentScale = ContentScale.Crop,
+    contentDescription: String? = null
+) {
+    Image(
+        painter = painterResource(imageId),
+        contentDescription = contentDescription,
+        contentScale = contentScale,
+        modifier = modifier
+            .clip(shape)
+    )
+}
```

Comment:
p3) 이 컴포넌트를 만든 이유가 있을까요?
단순 Image를 wrapping하기엔 Compose의 기본 컴포넌트를 하는거라 효과가 없을 것 같습니다.
물론 shape와 contentScale에 커스텀이 들어갔고, Spoony 서비스에서 대부분의 이미지에 해당옵션이 들어간다면 좋습니다 :)


p1) imageId를 받아오는 것 보다는 imageVector와 painter를 받아오는게 더욱 확장성이 좋을 것 같습니다.
DrawableRes를 받아오는 것도 좋은 방법이지만, imageVector와 painter를 구분하기 위해서는 직접 값을 받는게 더 깔끔할 것 같네요
```suggestion
@Composable
fun SpoonyImage(
    imageVecotr: ImageVector,
    modifier: Modifier = Modifier,
    shape: Shape = RectangleShape,
    contentScale: ContentScale = ContentScale.Crop,
    contentDescription: String? = null
) {
    Image(
        imageVector = imageVector,
        contentDescription = contentDescription,
        contentScale = contentScale,
        modifier = modifier
            .clip(shape)
    )
}

@Composable
fun SpoonyImage(
    painter: Painter,
    modifier: Modifier = Modifier,
    shape: Shape = RectangleShape,
    contentScale: ContentScale = ContentScale.Crop,
    contentDescription: String? = null
) {
    Image(
        painter = painter,
        contentDescription = contentDescription,
        contentScale = contentScale,
        modifier = modifier
            .clip(shape)
    )
}
```

Reply:
이 컴포넌트를 만든 이유는 SpoonyImage로 모든 타입의 이미지(url, drawable resource 등)를 처리할 수 있도록 하기 위함이었는데요! 다시 생각해보니 Image의 경우 wrapping하는 것에 이점이 전혀 없어서 AsyncImage만 만드는 것이 좋을 것 같습니다! 좋은 의견 감사합니다😊